packages/macros/src/with_components.rs

```diff
@@ -0,0 +1,857 @@
+/// * `has_initializer` - Whether the component requires an initializer (e.g. `true`)
+/// * `internal_impls` - The internal implementations of the component to be added to
+///   the module by default (e.g. `["InternalImpl1", "InternalImpl2"]`)
+#[derive(Debug, Clone)]
+pub struct ComponentInfo {
+    pub name: String,
+    pub path: String,
+    pub storage: String,
+    pub event: String,
+    pub has_initializer: bool,
+    pub has_immutable_config: bool,
+    pub internal_impls: Vec<String>,
+}
+
+impl ComponentInfo {
+    fn short_name(&self) -> String {
+        self.name
+            .split("Component")
+            .next()
+            .expect("Component name must end with 'Component'")
+            .to_string()
+    }
+}
+
+/// Returns the component info for a given component name.
+///
+/// # Arguments
+///
+/// * `name` - The name of the component (e.g. `ERC20`).
+///
+/// Allowed components are:
+/// `ERC20`, `Ownable`
+fn get_component_info(name: &str) -> (Option<ComponentInfo>, Diagnostics) {
+    let info = match name {
+        "Account" => Some(ComponentInfo {
+            name: "AccountComponent".to_string(),
+            path: "openzeppelin_account::AccountComponent".to_string(),
+            storage: "account".to_string(),
+            event: "AccountEvent".to_string(),
+            has_initializer: true,
+            has_immutable_config: false,
+            internal_impls: vec!["InternalImpl".to_string()],
+        }),
+        "EthAccount" => Some(ComponentInfo {
+            name: "EthAccountComponent".to_string(),
+            path: "openzeppelin_account::EthAccountComponent".to_string(),
+            storage: "eth_account".to_string(),
+            event: "EthAccountEvent".to_string(),
+            has_initializer: true,
+            has_immutable_config: false,
+            internal_impls: vec!["InternalImpl".to_string()],
+        }),
+        "SRC9" => Some(ComponentInfo {
+            name: "SRC9Component".to_string(),
+            path: "openzeppelin_account::extensions::SRC9Component".to_string(),
+            storage: "src9".to_string(),
+            event: "SRC9Event".to_string(),
+            has_initializer: true,
+            has_immutable_config: false,
+            internal_impls: vec!["InternalImpl".to_string()],
+        }),
+        "Ownable" => Some(ComponentInfo {
+            name: "OwnableComponent".to_string(),
+            path: "openzeppelin_access::ownable::OwnableComponent".to_string(),
+            storage: "ownable".to_string(),
+            event: "OwnableEvent".to_string(),
+            has_initializer: true,
+            has_immutable_config: false,
+            internal_impls: vec!["InternalImpl".to_string()],
+        }),
+        "AccessControl" => Some(ComponentInfo {
+            name: "AccessControlComponent".to_string(),
+            path: "openzeppelin_access::accesscontrol::AccessControlComponent".to_string(),
+            storage: "accesscontrol".to_string(),
```

Comment:
Same here. I know the lib isn't entirely consistent (might be worth addressing in a separate issue), but I think it could get weird when we have `timelock_controller` vs `accesscontrol` and `reentrancyguard`

Reply:
I will refactor them here (for macros). We may update presets and mocks in a separate issue later.